docs/faq.md

```diff
@@ -177,3 +177,32 @@ There is no golden rule, but you can follow these recommendations:
 
 This warning only occurs if you use non-default settings in your vite config that can cause problems in combination with prebundleSvelteLibraries.
 You should not use prebundleSvelteLibraries during build or for ssr, disable one of the incompatible options to make that warning (and subsequent errors) go away.
+
+<!-- the following header generates an anchor that is used in logging, do not modify!-->
+
+### deprecated "svelte" field
```

Comment:
per @dummdidumm's point, I'm not sure we need to actually deprecate the `svelte` field. we just need to stop resolving it before `exports` and resolve it after instead. that would make the warning trigger less frequently which would be nicer for users

Reply:
actually, I just looked at the code and I think it works assuming the svelte field will be left in place and only changes the order. if we really did want to get rid of it entirely we'd probably have to change the code, but I'd just suggest changing the title here and note below saying it will be removed and instead say the resolve order will change

Reply:
i don't think we can recover from a resolve error unless we _always_ call `this.resolve` in a try/catch  not something i want to do long term.

Reply:
also having 2 accepted ways to do this doesn't feel great. it makes writing new bundler plugins even more challanging. The only point i see in this whole excercise is that at the end of it the svelte field is no longer used at all.

Reply:
There's already two ways to resolve JS files with `main` and `exports`, so it's just the analogous thing. As long as we prefer `exports` we can rip out all the code in `vite-plugin-svelte` and just defer to Vite

Reply:
how do you expect
> **resolve it after instead.**

to happen after

> rip out all the code in vite-plugin-svelte and just defer to Vite

?

right now vite-plugin-svelte is a `pre` plugin exactly for the reason that we need to resolve via svelte field before vite resolve has a chance to error. Resolving the svelte field **after** a vite resolve error isn't possible unless you do in a pre-plugin. If you need to keep support for that alive after vite-plugin-svelte has switched it off, it should live in a legacy plugin that only users install that still have libraries in their project that would break otherwise. i don't want to burden regular use with that.

```js
// shortend 
async resolveId(importee,importer,options) {
  try {
    return this.resolve(importee,importer,{...options,skipSelf: true});
  } catch (e) {
    // vite resolve failed for any reason,
   const svelteResolved = await resolveViaPackageJsonSvelteField(importee,importer,options);
   if(svelteResolved) {
      log.warn('some warning here')
      return svelteResolved;
   }
   throw e;
  }
}

Reply:
`vite-plugin-svelte` wouldn't resolve it at all. We've set `svelte` in [`resolve.mainFields`](https://vitejs.dev/config/shared-options.html#resolve-mainfields), so Vite will handle it.

Reply:
I think we should phase out `pkg.svelte` too. `pkg.exports` is the official way to add custom condition names. Adding custom top-level fields isn't a standard and was only chosen because `pkg.exports` didn't exist then.

Reply:
> `vite-plugin-svelte` wouldn't resolve it at all. We've set `svelte` in [`resolve.mainFields`](https://vitejs.dev/config/shared-options.html#resolve-mainfields), so Vite will handle it.

no it won't. if exports entry exists, mainFields are ignored. mainFields also don't resolve non-js extensions afaik. There is no way to "fall back to svelte field" without wrapping vite resolve in a try catch - at least that i know of.

Reply:
> if exports entry exists, mainFields are ignored

I agree with that

> mainFields also don't resolve non-js extensions afaik

This would be news to me

Reply:
> mainFields also don't resolve non-js extensions afaik

I just tested this and it appears that Vite does resolve them. In order to test I installed a component (`svelte-simple-modal`), tweaked the `package.json` in the downloaded package to only have a `svelte` field pointing to a `.svelte` file, and then commented out `vite-plugin-svelte`'s call to `resolveViaPackageJsonSvelte`. It rendered the imported component just fine

Reply:
@bluwy  iirc you changed the behavior of exports + mainfields lately. Is the above still true? If yes we should be able to remove the custom code for resolveViaPackageJsonSvelte even today.

Reply:
can we change the title from `deprecated "svelte" field` to something more neutral like `prefer "svelte" export condition over "svelte" field`?

> kept the svelte field as neutral instead of remove. We should not encourage them to remove it for now as that could break older tooling that still relies on it

I agree with what you said in that comment and so the word "deprecated" seems too strong to me

Reply:
If `exports` is define, Vite should ignore `mainFields` (except `browser`) completely now. Vite also doesn't have a restriction on what the extensions for `mainFields` could be, so exporting a Svelte component is fine too.

Reply:
@dominikg any thoughts about the heading title here? I think it might be the only outstanding item for us to resolve

Reply:
> @dominikg any thoughts about the heading title here? I think it might be the only outstanding item for us to resolve

i have changed the wording to use conflicting resolve results instead of deprecation.